tools/rbac-gen/main.go

```diff
@@ -0,0 +1,104 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"io/ioutil"
+	"log"
+	"strings"
+	"sigs.k8s.io/kubebuilder-declarative-pattern/pkg/patterns/declarative/pkg/manifest"
+	"sigs.k8s.io/yaml"
+
+)
+
+type roleStruct struct {
+	Kind string `json:"kind"`
+	ApiVersion string `json:"apiVersion"`
+	Metadata metadata `json:"metadata"`
+	Rules []*rule `json:"rules"`
+}
+
+type metadata struct {
+	Name string `json:"name"`
+	Namespace string `json:"namespace"`
+}
+
+type rule struct {
+	ApiGroups []string `json:"apiGroups"`
+	Resources []string `json:"resources"`
+	Verbs []string `json:"verbs"`
+}
+
+var (
+	yamlFile = flag.String("yaml", "manifest.yaml", "yaml file from which the rbac will be generated.")
+	name = flag.String("name", "generated-role", "name of role to be generated")
+	ns = flag.String("ns", "kube-system", "namespace of the role to be generated")
+	out = flag.String("out.yaml", "kube-system", "name of output file")
+)
+
+func main() {
+	//	read yaml file passed in from cmd
+	flag.Parse()
+
+	bytes, err := ioutil.ReadFile(*yamlFile)
+	if err != nil {
+		log.Fatalf("Error reading files: %v", err)
```

Comment:
Tip: I like to do this:

```
func main() {
  err := run()
  if err != nil {
    fmt.Fprintf(os.Stderr, "%v\n", err)
    os.Exit(1)
  }
}

func run() error {
... real work....
}
```

because then we can just return an error.  Two advantages:

* A little more compact
* It makes it easier to refactor the code into functions; the top-level function is a special-case in terms of error handling, so I try to make that special-case as small as possible.

Reply:
This is nice. Reduce code for error handling